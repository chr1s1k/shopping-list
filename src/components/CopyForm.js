import React from 'react';

class CopyForm extends React.Component {
  render() {
    return (
      <p>Nyní už jen stačí zkopírovat adresu nákupu a někomu poslat!</p>
      <div className="action-zone form-group">
        <div className="input-group">
          <input type="text" name="shoppingListUrl" id="shoppingListUrl" className="form-control" value="/shopping-list-gui/public/" readonly="">
          <span className="input-group-btn">
            <button className="btn btn-primary" id="copyListUrlBtn" data-clipboard-target="#shoppingListUrl">Zkopírovat</button>
          </span>
        </div>
        <a href="" className="btn btn-link btn-block-xxs">Začít znova</a>
      </div>
    );
  }
}

export default CopyForm;