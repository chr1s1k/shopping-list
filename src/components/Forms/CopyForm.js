import React from 'react'
import PropTypes from 'prop-types'

class CopyForm extends React.Component {
	constructor() {
		super()

		this.handleOnClick = this.handleOnClick.bind(this)
		this.handleModifyList = this.handleModifyList.bind(this)
	}

	handleOnClick(event) {
		event.preventDefault()
		this.props.handleCreateNewList()
	}

	handleModifyList(event) {
		event.preventDefault()
		this.props.handleModifyList()
	}

	render() {
		return (
			<div>
				<div className="alert alert-success" role="alert">
					<p><i className="glyphicon glyphicon-ok"></i> Výborně! Nyní už stačí jen zkopírovat adresu nákupu a někomu poslat!</p>
				</div>
				<div className="action-zone form-group">
					<div className="input-group">
						<input type="text" name="shoppingListUrl" id="shoppingListUrl" className="form-control" value={this.props.listUrl} tabIndex="2" readOnly />
						<span className="input-group-btn">
							<button className="btn btn-primary" id="copyListUrlBtn" data-clipboard-target="#shoppingListUrl" tabIndex="1">Zkopírovat</button>
						</span>
					</div>
					<div className="btn-group" role="group">
						<button type="button" className="btn btn-link" onClick={this.handleModifyList}>Opravit nákup</button>
						<button type="button" className="btn btn-link" onClick={this.handleOnClick}>Napsat nový nákup</button>
					</div>
				</div>
			</div>
		)
	}
}

CopyForm.propTypes = {
	handleCreateNewList: PropTypes.func,
	handleModifyList: PropTypes.func,
	listUrl: PropTypes.string
}

export default CopyForm