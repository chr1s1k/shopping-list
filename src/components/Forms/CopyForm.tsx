import React, { MouseEvent } from 'react'

interface IProps {
	listUrl: string
	handleCreateNewList: () => void
	handleModifyList: () => void
}

class CopyForm extends React.Component<IProps> {
	constructor(props: IProps) {
		super(props)

		this.handleOnClick = this.handleOnClick.bind(this)
		this.handleModifyList = this.handleModifyList.bind(this)
	}

	handleOnClick(event: MouseEvent): void {
		event.preventDefault()
		this.props.handleCreateNewList()
	}

	handleModifyList(event: MouseEvent): void {
		event.preventDefault()
		this.props.handleModifyList()
	}

	render(): JSX.Element {
		return (
			<div>
				<div className="alert alert-success" role="alert">
					<p>
						<i className="glyphicon glyphicon-ok"></i> Výborně! Nyní už stačí
						jen zkopírovat adresu nákupu a někomu poslat!
					</p>
				</div>
				<div className="action-zone form-group">
					<div className="input-group">
						<input
							type="text"
							name="shoppingListUrl"
							id="shoppingListUrl"
							className="form-control"
							value={this.props.listUrl}
							tabIndex={2}
							readOnly
						/>
						<span className="input-group-btn">
							<button
								className="btn btn-primary"
								id="copyListUrlBtn"
								data-clipboard-target="#shoppingListUrl"
								tabIndex={1}
							>
								Zkopírovat
							</button>
						</span>
					</div>
					<div className="btn-group" role="group">
						<button
							type="button"
							className="btn btn-link"
							onClick={this.handleModifyList}
						>
							Opravit nákup
						</button>
						<button
							type="button"
							className="btn btn-link"
							onClick={this.handleOnClick}
						>
							Napsat nový nákup
						</button>
					</div>
				</div>
			</div>
		)
	}
}

export default CopyForm
