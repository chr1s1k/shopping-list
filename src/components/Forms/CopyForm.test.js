import React from 'react'
import { shallow } from 'enzyme'

import CopyForm from './CopyForm'

describe('CopyForm', () => {
	it('should render correctly without props', () => {
		const component = shallow(<CopyForm />)

		expect(component).toMatchSnapshot()
	})

	it('should render correctly with listURL prop', () => {
		const component = shallow(<CopyForm listUrl="/test/url" />)

		expect(component).toMatchSnapshot()
	})
})
