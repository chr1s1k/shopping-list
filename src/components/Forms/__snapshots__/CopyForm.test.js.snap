// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CopyForm should render correctly with listURL prop 1`] = `
<div>
  <div
    className="alert alert-success"
    role="alert"
  >
    <p>
      <i
        className="glyphicon glyphicon-ok"
      />
       Výborně! Nyní už stačí jen zkopírovat adresu nákupu a někomu poslat!
    </p>
  </div>
  <div
    className="action-zone form-group"
  >
    <div
      className="input-group"
    >
      <input
        className="form-control"
        id="shoppingListUrl"
        name="shoppingListUrl"
        readOnly={true}
        tabIndex="2"
        type="text"
        value="/test/url"
      />
      <span
        className="input-group-btn"
      >
        <button
          className="btn btn-primary"
          data-clipboard-target="#shoppingListUrl"
          id="copyListUrlBtn"
          tabIndex="1"
        >
          Zkopírovat
        </button>
      </span>
    </div>
    <div
      className="btn-group"
      role="group"
    >
      <a
        className="btn btn-link"
        href=""
        onClick={[Function]}
        role="button"
      >
        Opravit nákup
      </a>
      <a
        className="btn btn-link"
        href=""
        onClick={[Function]}
        role="button"
      >
        Napsat nový nákup
      </a>
    </div>
  </div>
</div>
`;

exports[`CopyForm should render correctly without props 1`] = `
<div>
  <div
    className="alert alert-success"
    role="alert"
  >
    <p>
      <i
        className="glyphicon glyphicon-ok"
      />
       Výborně! Nyní už stačí jen zkopírovat adresu nákupu a někomu poslat!
    </p>
  </div>
  <div
    className="action-zone form-group"
  >
    <div
      className="input-group"
    >
      <input
        className="form-control"
        id="shoppingListUrl"
        name="shoppingListUrl"
        readOnly={true}
        tabIndex="2"
        type="text"
      />
      <span
        className="input-group-btn"
      >
        <button
          className="btn btn-primary"
          data-clipboard-target="#shoppingListUrl"
          id="copyListUrlBtn"
          tabIndex="1"
        >
          Zkopírovat
        </button>
      </span>
    </div>
    <div
      className="btn-group"
      role="group"
    >
      <a
        className="btn btn-link"
        href=""
        onClick={[Function]}
        role="button"
      >
        Opravit nákup
      </a>
      <a
        className="btn btn-link"
        href=""
        onClick={[Function]}
        role="button"
      >
        Napsat nový nákup
      </a>
    </div>
  </div>
</div>
`;
